packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js

```diff
@@ -48,4 +58,23 @@ export default class OvhPaymentMethodIntegrationInContextPaypalCtrl {
       .then(({ formSessionId }) => formSessionId)
       .catch(() => {});
   }
+
+  onButtonsInit(_, buttonsCtrl) {
+    this.buttonsCtrl = buttonsCtrl;
+    if (this.inContextCtrl.integrationCtrl.disabled) {
+      this.disableButtons();
+    }
+  }
+
+  disableButtons() {
+    if (this.buttonsCtrl) {
+      this.buttonsCtrl.disable();
+    }
+  }
```

Comment:
To reduce code, you can group enable and disable state in one function
```suggestion
  manageButtonsState(disabled = true) {
    if (this.buttonsCtrl) {
      disabled ? this.buttonsCtrl.disable() :  this.buttonsCtrl.enable() ;
    }
  }
```

Reply:
I disagree, this lacks readability, you have to look into `manageButtonsState` and pay attention to the `disabled` argument to know what the function does. To me it's not worth it just for a few lines of code. In the other hands `disableButtons` and `enableButtons` are pretty self explanatory

Reply:
Well, it was just an example but you can rename it for more readability 
```suggestion
  activateButtons(disabled) {
         if (this.buttonsCtrl) {
      disabled ? this.buttonsCtrl.disable() :  this.buttonsCtrl.enable() ;
    }
    }
  }
```